docs/source/spec/xml.rst

```diff
@@ -1,26 +1,598 @@
+serialized in a nested element named ``member``. For example, given the
+following:
+
+.. code-block:: smithy
+
+    structure Foo {
+        values: MyList
+    }
+
+    list MyList {
+        member: String,
+    }
+
+The XML serialization of ``Foo`` is:
+
+.. code-block:: xml
+
+    <Foo>
+        <values>
+            <member>example1</member>
+            <member>example2</member>
+            <member>example3</member>
+        </values>
+    </Foo>
+
+The :ref:`xmlname-trait` can be applied to the member of a list or set to
+change the nested element name. For example, given the following:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        structure Foo {
+            values: MyList
+        }
+
+        list MyList {
+            @xmlName("Item")
+            member: String,
+        }
+
+    .. code-tab:: json
+
+        {
+            "smithy": "0.5.0",
+            "shapes": {
+                "smithy.example#Foo": {
+                    "type": "structure",
+                    "members": {
+                        "values": {
+                            "target": "smithy.example#MyList"
+                        }
+                    }
+                },
+                "smithy.example#MyList": {
+                    "type": "list",
+                    "member": {
+                        "target": "smithy.api#String",
+                        "traits": {
+                            "smithy.api#xmlName": "Item"
+                        }
+                    }
+                }
+            }
+        }
+
+The XML serialization of ``Foo`` is:
+
+.. code-block:: xml
+
+    <Foo>
+        <values>
+            <Item>example1</Item>
+            <Item>example2</Item>
+            <Item>example3</Item>
+        </values>
+    </Foo>
+
+
+Flattened list serialization
+----------------------------
+
+The :ref:`xmlflattened-trait` can be used to unwrap the values of list/set
+into a containing structure/union. The name of the elements repeated within
+the structure/union is based on the structure/union member name. For
+example, given the following:
+
+.. code-block:: smithy
+
+    structure Foo {
+        @xmlFlattened
+        flat: MyList,
+    }
+
+The XML serialization of ``Foo`` is:
+
+.. code-block:: xml
+
+    <Foo>
+        <flat>example1</flat>
+        <flat>example2</flat>
+        <flat>example3</flat>
+    </Foo>
+
+The ``xmlName`` trait applied to the structure/union member is used to change
+the name of the repeated XML element. For example, given the following:
+
+.. code-block:: smithy
+
+    union Choice {
+        @xmlFlattened
+        @xmlName("Hi")
+        flat: MySet,
+    }
+
+    set MySet {
+        member: String
+    }
+
+The XML serialization of ``Choice`` is:
+
+.. code-block:: xml
+
+    <Choice>
+        <Hi>example1</Hi>
+        <Hi>example2</Hi>
+        <Hi>example3</Hi>
+    </Choice>
+
+The ``xmlName`` trait applied to the member of a list has no effect when
```

Comment:
Nit: Should this paragraph use "list/set" like the opening paragraph, since "structure/union" is preserved as well? The proceeding section uses "list or set", which is different as well.